### DISCUSSION TITLE
Watch Data Change or Data Binding in Go?

### DISCUSSION THREAD

**Discussion by KiddoV** [wailsapp/wails#1730] [Ideas]
Would be nice to be able do something like this in ``wails``

https://siongui.github.io/2018/05/04/watch-data-change-via-gopherjs-vue/

So when we change something in ``go`` models, the view in front-end will updates accordingly.

Would this be something that you can implement?

**Comment by leaanthony** [comment]
V1 had the concept of stores for a short while but there were issues with concurrency due to the JSON encoder panicking when the value was updated concurrently or someone modified the data outside of the store. 

If you look carefully, the example given is not in Go - it's gopherjs and purely frontend.

Now that generics are here, I think we can relook at this. I actually thought there was already a ticket for it.

**Reply by KiddoV** [reply]
Thanks for the reply @leaanthony , I just want to bring this up to see if you can implement it. It would be a nice feature to have.
Do you want me to create an ``issue`` for this?
